### PR TITLE
[release-v0.24] Use dependencies instead of IMAGE_FORMAT and switch to cluster pool

### PR DESCRIPTION
**Pull request by devguyio**
- Switch to dependencies instead of IMAGE_FORMAT
- Use cluster pool for 4.8

based on  openshift/knative-eventing#1377

### CODE REVIEW

**Comment by matzew** [comment]
We should update the generator for the CI file:

```patch
diff --git a/openshift/ci-operator/generate-ci-config.sh b/openshift/ci-operator/generate-ci-config.sh
index 4a1e52f44..312df20c4 100755
--- a/openshift/ci-operator/generate-ci-config.sh
+++ b/openshift/ci-operator/generate-ci-config.sh
@@ -40,10 +40,16 @@ EOF
 image_deps=$(print_image_dependencies)
 
 cat <<EOF
-tag_specification:
-  cluster: https://api.ci.openshift.org
-  name: '$openshift'
-  namespace: ocp
+releases:
+  initial:
+    integration:
+      name: '$openshift'
+      namespace: ocp
+  latest:
+    integration:
+      include_built_images: true
+      name: '$openshift'
+      namespace: ocp
 promotion:
   additional_images:
     knative-eventing-kafka-src: src
```


(this change needs to go to a few more branches)

**Comment by matzew** [comment]
A reference PR: https://github.com/openshift/knative-eventing/pull/1406   

(we can do that separately too... does not hurt) 

Rest of the change looks good

**Comment by matzew** [comment]
> Use cluster pool for 4.8

why only 4.8 ? 

If we do not want 4.7 for this (fine with me), lets update this:
https://github.com/devguyio/eventing-kafka/blob/release-v0.24/openshift/ci-operator/update-ci.sh#L40-L41

```
$CURDIR/generate-ci-config.sh knative-$VERSION 4.8 > ${CONFIG}__48.yaml
```

and we than remove the 4.7 run config on `openshift/release`

**Comment by mgencur** [comment]
We only add cluster pools for 4.8 because of a bug in Hive which causes failures with cluster hibernation. We don't hit this bug on OCP 4.8 because it's frequently used by other teams and clusters are rarely hibernated.

File: openshift/ci-operator/generate-ci-config.sh [hunk 1]
```diff
@@ -32,13 +63,64 @@ canonical_go_repository: knative.dev/eventing-kafka
 binary_build_commands: make install
 test_binary_build_commands: make test-install
 tests:
+EOF
+if [[ "$openshift" == "4.8" ]]; then
```
Comment (matzew) [inline openshift/ci-operator/generate-ci-config.sh:67]: check :wink:

**Comment by matzew** [comment]
```
--- FAIL: TestKafkaBinding/structured-v1beta1 (2.39s)
panic: could not parse image template: template: image:1: illegal number syntax: "-" [recovered]
	panic: could not parse image template: template: image:1: illegal number syntax: "-"
```


looks like something wrong w/ the ENV vars ? 

(or are they not yet live, via OCP/release repo merge)

/cc @mgencur

File: openshift/e2e-tests.sh [hunk 2]
```diff
@@ -6,7 +6,7 @@ source "$(dirname "$0")/e2e-common.sh"
 
 set -Eeuox pipefail
 
-export TEST_IMAGE_TEMPLATE="${IMAGE_FORMAT//\$\{component\}/knative-eventing-kafka-test-{{.Name}}}"
+export TEST_IMAGE_TEMPLATE="$EVENTING_KAFKA_TEST_IMAGE_TEMPLATE"
```
Comment (matzew) [inline openshift/e2e-tests.sh:9]:
```
export TEST_IMAGE_TEMPLATE="${EVENTING_KAFKA_TEST_IMAGE_TEMPLATE}"
```

perhaps?

Comment (mgencur) [reply openshift/e2e-tests.sh:9]: Yes. I think it should be changed here.

**Review by mgencur — COMMENTED** [review]
@matzew they should be live now. We can try to re-run the check after fixing the issue from https://github.com/openshift-knative/eventing-kafka/pull/342#discussion_r709172792

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Pull request merged**